### ISSUE TITLE
Private logger property

### ISSUE THREAD

**Issue by marghidanu** [dot-i/k8s-operator-node#20]
It would be nice if the logger property would be `protected` instead of `private`; that would allow its use in the `init` function.

**Comment by dot-i** [comment]
You pass in the logger yourself in the constructor, so you should already have it available in the first place, right? 🙂

**Comment by marghidanu** [comment]
Not true, the property is marked as private, therefore is not accessible in subclasses.

**Comment by dot-i** [comment]
You misunderstand me.

You are supposed to pass the logger like this in the constructor of your class:

```js
  constructor() {
    super(this.logger);
  }
````

So what I mean is that you obviously already need to have the logger available. If you can pass it to `super()` you can also access it for other logging purposes.

**Comment by marghidanu** [comment]
That's exactly what I'm doing, and then I'm trying to use `this.logger` inside of the newly defined init function. and the property logger is not available because is private (not protected).

**Comment by dot-i** [comment]
I should have shown the code like this, as I caused further confusion with the `logger` property on the base class (my bad):

```js
  constructor() {
    super(this.yourLogger);
  }
```

You understand my point now?  I can still make the property protected, that's not really that big a deal, but it seems a bit strange to me that it should block you, as you are the owner of the logger you passed in to the `super()` call in the first place 🙂

**Comment by marghidanu** [comment]
I understand perfectly what you are saying, but that means that I have an unwanted dependency that is not on my class; let me give you an example. I have in my `operator.ts` an implementation of the Operator:

```ts
import Operator from '@dot-i/k8s-operator';

export default class MyOperator extends Operator {
    protected async init(): Promise<void> {
        this.logger.info("default logger"); // Property 'logger' is private and only accessible within class 'Operator'.
    }
}
```
In my `index.ts` file I have this:

```ts
import winston from 'winston';
import MyOperator from './components/operator2';

const logger = winston.createLogger({
    level: 'info',
    transports: [new winston.transports.Console()],
});

const operator = new MyOperator(logger);
operator.start();
```

In my case, the logger is shared between other components (like a web component); I have only one definition for the logger. If the property would be `protected`, I wouldn't need to export the logger from anywhere, and I wouldn't need to override the constructor of my operator class.  Also, if the attribute is already in the class why not use it? :)

Does it make sense?

**Comment by dot-i** [comment]
I just published 1.3.3 which makes the logger protected.
